<?xml version="1.0"?>
<launch>
    <arg name="bag" /> <!-- bag:=path/to/bag.bag -->
    <arg name="path" default="$(find recorder)/data/$(anon dataset)" />
    <!--<arg name="delay" value="10"/>-->
    
    <param name ="/use_sim_time" value="true"/>
    <!--<param name="/recorder/delay" value="$(arg delay)"/>-->
    
    <!-- TODO: An init script to setup the cameras and create the folder (remove from TF rec) -->
   
    <node pkg="rosbag" type="play" name="rosbag" args="--clock -d 0.0 $(arg bag)" required="true"/>

    <include file="$(find recorder)/launch/record_tf.launch">   <!-- The TF recorder also creates the folders -->
        <arg name="path" value="$(arg path)" />
    </include>
    
    <node name="extract_left" pkg="image_view" type="extract_images" respawn="false" output="log" cwd="ROS_HOME">
        <remap from="image" to="/cameras/left_hand_camera/image"/>
        <param name="sec_per_frame" value="0.001"/>
        <param name="filename_format" value="left%d.jpg"/>
    </node>
    
    <node name="extract_right" pkg="image_view" type="extract_images" respawn="false" output="log" cwd="ROS_HOME">
        <remap from="image" to="/cameras/right_hand_camera/image"/>
        <param name="sec_per_frame" value="0.001"/>
        <param name="filename_format" value="right%d.jpg"/>
    </node>
    
    <node name="extract_head" pkg="image_view" type="extract_images" respawn="false" output="log" cwd="ROS_HOME">
        <remap from="image" to="/usb_cam/image_raw"/>
        <param name="sec_per_frame" value="0.001"/>
        <param name="filename_format" value="head%d.jpg"/>
    </node>
    
</launch>
